tycho-packaging-plugin/src/main/java/org/eclipse/tycho/buildversion/BuildQualifierMojo.java

```diff
@@ -175,7 +175,9 @@ private TychoProjectVersion calculateQualifiedVersion(Date timestamp)
         }
 		String qualifier = getDesiredQualifier(timestamp);
 
-        validateQualifier(qualifier);
+		if (!"".equals(qualifier)) {
```

Comment:
Can this check be performed in `validateQualifier`?

Reply:
I reworked according our discussion to use `forceContextQualifier=none` and moved checks inside, preserving flow in calculate method. Let me know if it looks fine to you, I can make needed adjustments if necessary.